pkg/operator/ceph/pool/controller.go

```diff
@@ -260,3 +243,28 @@ func deletePool(context *clusterd.Context, p *cephv1.CephBlockPool) error {
 
 	return nil
 }
+
+// UpdateStatus updates a pool CR with the given status
+func updateStatus(client client.Client, poolName types.NamespacedName, status string) {
```

Comment:
> I'm not a big fan of decoupling the centralized helper into each controller implementation.

@leseb Is this the helper you're referring to? Let's chat about it...

Reply:
Yes! Let's chat about this.